R/mbbs.R

```diff
@@ -107,12 +133,45 @@ create_stop_level_counts <- function(ebird, taxonomy, config = config) {
 create_route_level_counts_0 <- function(ebird, stop_level_data, taxonomy, config = config) {
   logger::log_trace("Computing route level data from stop level")
   stop_to_route <- stop_level_data |>
-    group_by(year, county, route, route_num, common_name, sci_name) |>
+    group_by(year, county, route, route_num, common_name, sci_name, source) |>
     summarise(count = sum(count)) |>
     select(
-      year, common_name, sci_name, route, route_num, county, count
+      year, common_name, sci_name, route, route_num, county, count, source
     )
 
+  # Compare stop_level and ebird counts
+  dplyr::left_join(
+    stop_to_route |> dplyr::ungroup() |>
+      dplyr::select(common_name, year, route, scount = count, source),
+    ebird |> dplyr::filter(is.na(stop_num)) |> dplyr::ungroup() |>
+      dplyr::select(common_name, year, route, rcount = count),
+    by = c("common_name", "year", "route")
+  ) |>
+    dplyr::mutate(diff = scount - rcount) |>
+    (\(x) {
+      logger::log_trace(
+        paste(
+          "Comparing {nrow(x)}",
+          "stop-to-route observations with ebird (without stops)"
+        )
+      )
+      x
+    })() |>
+    dplyr::filter(diff != 0) |>
+    (\(x) {
+      if (nrow(x) > 0) {
+        logger::log_warn(
+          paste(
+            "{x$year}-{x$route} had",
+            "{x$scount} {x$common_name} aggregrated in stop_level",
+            "{x$source}",
```

Comment:
Adding 'source' here and lines 135 are the changes in this section from the other pull request